### PR TITLE
Check types before skipping props

### PR DESCRIPTION
**Pull request by slaskis**
This is a PR to try to solve #59. 

The test is very simple (just added a `properties` field to the fixture) so it may miss some cases.

The change really only handles geometries for GeometryCollections and properties for Features.

### CODE REVIEW

File: encode.js [hunk 1]
```diff
@@ -138,12 +138,13 @@ function writeProps(props, pbf, isCustom) {
 
     for (var key in props) {
         if (isCustom) {
-            if (key === 'type' || key === 'id' || key === 'coordinates' || key === 'arcs' ||
-                key === 'geometries' || key === 'properties') continue;
```
Comment (slaskis) [inline encode.js]: Is there a case you are aware of that requires skipping the `properties` key?

**Comment by mourner** [comment]
It looks like this code wasn't properly cleaned up after we removed TopoJSON support, where geometries could have properties, etc. I think we should change the code to be as specific as possible — e.g. only skip a key if it's already encode as a part of non-custom writeProps (properties for feature, geometries for geometry collection, etc.)

**Comment by mourner** [comment]
For some reason Travis was disabled for the repo, — I reenabled it and it should build the next commit you push.

**Comment by slaskis** [comment]
Should the same check be updated for the analysis phase? 

I'm talking about these lines: https://github.com/slaskis/geobuf/blob/fc-properties/encode.js#L70-L73

**Comment by slaskis** [comment]
I've updated where I don't bother with the TopoJSON `arcs` key and only check for the `coordinates` if "other" type, which I'm assuming implicitly should always be a geometry type.

File: encode.js [hunk 2]
```diff
@@ -138,13 +138,14 @@ function writeProps(props, pbf, isCustom) {
...
+            if (key === 'type' || key === 'id') continue;
```
Comment (mourner) [inline encode.js]: `id` is now only encoded for features

**Comment by mourner** [comment]
This looks good! No need to be overly explicit I think. We should also update analyse routine and ideally reuse the code above — maybe cut it out into a `isSpecialKey(key, type)` function used in both places?

**Comment by slaskis** [comment]
On it!

**Comment by slaskis** [comment]
Allright, much nicer in a little function. I also fixed your inline-comment regarding the `id` key.

**Comment by mourner** [comment]
@slaskis nice! Still some properties hardcoded in `analyse`, should use `isSpecialKey` instead.

**Comment by slaskis** [comment]
Good catch.

I moved the properties loop out of the ifs. And the tests still pass.

**Comment by slaskis** [comment]
I've notices there are now switch statements in the code base. Is this an intentional code style?

Because personally I'd think the `analyze()` function would be a bit easier to read like this:

```
function analyze(obj) {
    var i, key;

    switch (obj.type) {
        case 'FeatureCollection':
            for (i = 0; i < obj.features.length; i++) analyze(obj.features[i]);
            break;
        case 'Feature':
            analyze(obj.geometry);
            for (key in obj.properties) saveKey(key);
            break;
        case 'Point':
            analyzePoint(obj.coordinates);
            break;
        case 'MultiPoint':
        case 'LineString':
            analyzePoints(obj.coordinates);
            break;
        case 'Polygon':
        case 'MultiLineString':
            analyzeMultiLine(obj.coordinates);
            break;
        case 'GeometryCollection':
            for (i = 0; i < obj.geometries.length; i++) analyze(obj.geometries[i]);
            break;
        case 'MultiPolygon':
            for (i = 0; i < obj.coordinates.length; i++) analyzeMultiLine(obj.coordinates[i]);
            break;

    }

    for (key in obj) {
        if (!isSpecialKey(key, obj.type)) saveKey(key);
    }
}
```

**Comment by mourner** [comment]
@slaskis just a personal preference. The PR looks good! Can you squash into one commit?

**Comment by slaskis** [comment]
Good to go!

**Comment by mourner** [comment]
Great, thanks!

**Pull request merged**